subworkflows/local/differential/main.nf

```diff
@@ -0,0 +1,86 @@
+//
+// Perform differential analysis
+//
+include { PROPR_PROPD as PROPD } from "../../../modules/local/propr/propd/main.nf"
+include { DESEQ2_DIFFERENTIAL  } from '../../../modules/nf-core/deseq2/differential/main'
+
+workflow DIFFERENTIAL {
+    take:
+    ch_tools
```

Comment:
If I understand correctly your point, you want to parse the 'diff_method' and 'args_diff' meta values into one 'diff' channel?
That is actually what we aim to do with the next changes for 'ch_tools'. 
Currently, all the metas for different subworkflows are parsed inside 'ch_tools' and provided to each subworkflow.
But we want to modify it so that, 
   - differential subworkflow will only be provided with a ch_tools that see the metas 'diff_method' and 'args_diff'.
   - correlation subworkflow will only have a ch_tools that see 'cor_method' and 'args_cor'.
   - enrichment subworkflow will have a ch_tools that see everything ('diff_method', 'args_diff', 'cor_method', 'args_cor', 'enr_method', 'args_enr'). The reason is that the input coming from upstream will be different and they are characterized by their corresponding 'diff_method' 'args_diff' or 'cor_method' 'args_cor'.

The advantage to still have a 'ch_tools' inside the subworkflow are many:
  - we can easily use the information from 'ch_tools' to run one or more processes inside the same subworkflow.
  - We can parse the 'ch_tools' with its corresponding input data (ch_counts, ch_contrast, etc) in the way how each module requires, just before running the module. So it is visually clearer to understand what kind of parsing, etc are going on for a given module. I think in this way it is easier to understand what goes with which module, and hence also easier in the future to add new modules. For the new addition, one just have to consider to parse the inputs from the subworkflow with the channel just before calling this new module.

Reply:
What I mean is, we **definitely** shouldn't have things like:

```
ch_tools
        .branch {
            propd:  it[0]["diff_method"] == "propd"
            deseq2: it[0]["diff_method"] == "deseq2"
        }
        .set { ch_tools_single }
```

That requires the subworkflow to know things about the calling suborkflow. The workflow should just `take:` the diff_method. 

We can use `multiMap` outside of the subworkflow to generate the right channels:

```
ch_tools
    .multiMap { it ->
        ch_diff_method: it[0]["diff_method"]
        ch_other_option: it[0]["other_option"]
    }
    .set { result }
```

In the above example, ch_diff_method and ch_other option will maintain their relative order, so could be passed directly to DIFFERENTIAL. 

By doing things that way, the worklow has 'encapsulation'. You pass it the things it needs, and it doesn't need to know anything about what's done outside of the subworkflow.